lib/galaxy/tools/__init__.py

```diff
@@ -2352,8 +2347,8 @@ def to_json (self, trans, kwd={}, is_workflow=False):
         tool_message = ''
         if job:
             try:
-                job_params = job.get_param_values( trans.app, ignore_errors = True )
-                job_messages = self.check_and_update_param_values( job_params, trans, update_values=False )
+                job_params = job.get_param_values( trans.app, ignore_errors=True )
+                self.check_and_update_param_values( job_params, trans, update_values=False )
```

Comment:
Ahh, yeah, you're right; it should be modified in the helper.  So, back to the original statement; instead of dropping the value I'd append it to the return json as it was probably intended to be used.